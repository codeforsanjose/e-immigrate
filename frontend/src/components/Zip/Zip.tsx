import React from 'react';
import './Zip.css';
import { ReactSetter } from '../../types/common';
import { CommonComponentProps } from '../../types/CommonComponentProps';
import { useContentContext } from '../../contexts/ContentContext';
import { RequiredErrorDiv } from '../RequiredErrorPresenter';
import { useQuestionnaireResponseContext } from '../../contexts/QuestionnaireResponseContext';
type ZipProps = CommonComponentProps & {
    setErrors: ReactSetter<Record<string, unknown>>;
};
export function Zip(props: ZipProps) {
    const {
        q, 
        setErrors, 
    } = props;
    const { content } = useContentContext();
    const { 
        collectAnswer,
        bindField, 
    } = useQuestionnaireResponseContext();
    return (
        <>
            <input
                type="text"
                pattern="[0-9]{5}"
                id={q.slug}
                name={q.slug}
                required={q.required}
                className="TextInput"
                {...bindField(q.slug)}
                onChange={(e) => {
                    const state = e.target.checkValidity();
                    // TODO : NEED TO FIX error setting, check validity is working properly but error flag is not propergating
                    setErrors(prev => {
                        if (prev[q.slug] === state) return prev;
                        return {
                            ...prev,
                            [q.slug]: state,
                        };
                    });
                    collectAnswer(q.slug, e.target.value);
                } } />
            <RequiredErrorDiv message={content.errorMessageZip} />
        </>
    );
}
