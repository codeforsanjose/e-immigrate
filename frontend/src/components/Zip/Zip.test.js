import React from 'react';
import Zip from './Zip';

import renderer from 'react-test-renderer';

it('Zip renders correctly', () => {
    const tree = renderer
        .create(
            <Zip
                q={{}}
                answers={[]}
                values={[]}
                bindField={() => {}}
                showFollowUp={{}}
                setShowFollowUp={{}}
                collectAnswer={{}}
                content={{}}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
