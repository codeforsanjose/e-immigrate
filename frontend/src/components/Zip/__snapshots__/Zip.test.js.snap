// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zip renders correctly 1`] = `
Array [
  <input
    className="TextInput"
    onChange={[Function]}
    pattern="[0-9]{5}"
    type="text"
  />,
  <div
    className="RequiredError"
  >
    *
  </div>,
]
`;
